library/std/src/sys/unix/mod.rs

```diff
@@ -60,25 +58,23 @@ pub fn init() {
     //
     // Hence, we set SIGPIPE to ignore when the program starts up in order
     // to prevent this problem.
-    unsafe {
-        reset_sigpipe();
-    }
-
-    cfg_if::cfg_if! {
-        if #[cfg(miri)] {
-            // The standard fds are always available in Miri.
-            unsafe fn sanitize_standard_fds() {}
-        } else if #[cfg(not(any(
-            target_os = "emscripten",
-            target_os = "fuchsia",
-            // The poll on Darwin doesn't set POLLNVAL for closed fds.
-            target_os = "macos",
-            target_os = "ios",
-            target_os = "redox",
-        )))] {
-            // In the case when all file descriptors are open, the poll has been
-            // observed to perform better than fcntl (on GNU/Linux).
-            unsafe fn sanitize_standard_fds() {
+    reset_sigpipe();
+
+    stack_overflow::init();
+    args::init(argc, argv);
+
+    unsafe fn sanitize_standard_fds() {
+        cfg_if::cfg_if! {
```

Comment:
This `cfg_if` doesn't result in the same behaviour as before on `miri` + `target_os = "macos"`.